crates/bevy_ecs/src/system/commands.rs

```diff
@@ -189,17 +190,50 @@ pub struct Commands {
 }
 
 impl Commands {
+    /// Creates a new entity and calls `insert` with the it and `components`.
+    ///
+    /// Note that `components` is a bundle. If you would like to spawn an entity with a single component, consider wrapping the component in a tuple (which `DynamicBundle` is implemented for).
+    ///
+    /// See `set_current_entity`, `insert`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use bevy_ecs::prelude::*;
+    ///
+    /// struct Component1;
+    /// struct Component2;
+    ///
+    /// #[derive(Bundle)]
+    /// struct ExampleBundle {
+    ///     a: Component1,
+    ///     b: Component2,
+    /// }
+    ///
+    /// fn example_system(mut commands: Commands) {
+    ///     // Create a new entity with a component bundle.
+    ///     commands.spawn(ExampleBundle {
+    ///         a: Component1,
+    ///         b: Component2,
+    ///     });
+    ///
+    ///     // Create new entities with a single component each.
+    ///     commands.spawn((Component1,));
+    ///     commands.spawn((Component2,));
+    /// }
+    /// ```
     pub fn spawn(&mut self, components: impl DynamicBundle + Send + Sync + 'static) -> &mut Self {
         let entity = self
             .entity_reserver
             .as_ref()
             .expect("entity reserver has not been set")
             .reserve_entity();
-        self.current_entity = Some(entity);
-        self.commands.push(Box::new(Insert { entity, components }));
+        self.set_current_entity(entity);
+        self.insert(entity, components);
         self
     }
 
+    /// Equivalent to iterating of `components_iter` and calling `spawn` on each bundle, but slightly more performant.
```

Comment:
Is this correct? I'm basing this off of the method signature, but `World::spawn_batch`'s docs seem to imply that `I` should be an iterator of `Component` rather than `Bundle`

Reply:
@cart I wonder if you can help clarify this for me?